src/toil/common.py

```diff
@@ -1268,6 +1273,36 @@ def getToilWorkDir(configWorkDir: Optional[str] = None) -> str:
             raise RuntimeError(f'The directory specified by --workDir or TOIL_WORKDIR ({workDir}) does not exist.')
         return workDir
 
+    @staticmethod
+    def get_working_tmpdir(tmpdir_prefix: Optional[str] = None) -> Optional[str]:
+        """
+        Get a working temp directory, testing if it is accessible
+        If tmpdir_prefix is given, assuming it is accessible, it will return its parent directory
+        if the path itself is not a directory path
+        :param tmpdir_prefix: tmp directory to override
+        :return: Path or none
+        """
+        # if tmpdir_prefix exists, return its parent or None depending on if it is accessible
+        if tmpdir_prefix is not None:
+            return tmpdir_prefix if os.path.isdir(tmpdir_prefix) else os.path.dirname(tmpdir_prefix)
+
+        # Priority will be: TMPDIR > TEMP > TMP > /tmp
+        # gettempdir returns the current working directory as a last resort
+        # but we don't want that, so return None if the cwd is reached
```

Comment:
What if `TMPDIR` is `/tmp` and we happen to be standing in `/tmp` when running the workflow? I don't think we want to disqualify a directory *just because* it's the current directory. If it has to fall all the way back to the current directory as our temp directory, maybe we should in fact use it as our temp directory?

Reply:
Would we want to prioritize Toil's work dir over the current working directory? If I don't omit the current directory from `gettempdir`, then I think Toil's work dir will never be reached, as gettempdir() should always be returning something (unless toil is somehow installed in a nonwritable directory).

Reply:
It would be nice if we could prioritize the workDir below actual temp directories like the ones from the environment variables and `/tmp`, but above the current working directory as a fallback if `/tmp` is unavailable.

But if the current working directory *is itself `/tmp`*, we still want to use `/tmp`.